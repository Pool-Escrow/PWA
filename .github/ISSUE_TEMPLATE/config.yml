blank_issues_enabled: false

contact_links:
  - name: 💬 Community Slack Channel
    url: https://tinyurl.com/pool-escrow-slack
    about: Join our Slack for quick help and community discussions! 🤝

  - name: 🎮 Community Discord Server
    url: https://discord.gg/pool-escrow
    about: Join our Discord for real-time chat and support! 🎉

  - name: 📚 Documentation
    url: https://docs.pool-escrow.com
    about: Check our documentation for guides and FAQs. 📖

  - name: 🐦 Twitter Support
    url: https://twitter.com/PoolEscrowHelp
    about: Follow us on Twitter for updates and quick support! 🐤

  - name: 📘 Contributing Guidelines
    url: https://github.com/your-org/your-repo/blob/main/CONTRIBUTING.md
    about: Please read our contributing guidelines before submitting an issue or PR.
