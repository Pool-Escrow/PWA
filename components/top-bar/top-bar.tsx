import { TopBarButton } from './top-bar-button'
import TopBarLogo from './top-bar-logo'

interface TopBarProps {
	left?: React.ReactNode
	center?: React.ReactNode
	right?: React.ReactNode
}

export default function TopBar({ left, center, right }: TopBarProps) {
	return (
		<header className='fixed left-0 top-0 w-full bg-white h-20 z-10'>
			<nav className='mx-auto flex h-full max-w-screen-md items-center justify-around px-6'>
				<div className='flex-1 flex justify-start'>{left}</div>
				<div>{center}</div>
				<div className='flex-1 flex justify-end'>{right}</div>
			</nav>
		</header>
	)
}

TopBar.Button = TopBarButton
TopBar.Logo = TopBarLogo
