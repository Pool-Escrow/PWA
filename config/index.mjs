// @ts-check

import { imageConfig } from './images.mjs'
import { getRewriteRules } from './rewrites.mjs'
import { getSecurityHeaders } from './security.mjs'
import { configureWebpack } from './webpack.mjs'
import { withSerwist } from './serwist.mjs'
import { compilerConfig } from './compiler.mjs'
import { experimentalConfig } from './experimental.mjs'

const config = {
    images: imageConfig,
    rewrites: getRewriteRules,
    security: getSecurityHeaders,
    webpack: configureWebpack,
    serwist: withSerwist,
    compiler: compilerConfig,
    experimental: experimentalConfig,
}

export default config
