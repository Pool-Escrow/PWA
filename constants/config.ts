import { createConfig, http } from 'wagmi'
import { foundry } from 'wagmi/chains'

export const config = createConfig({
	chains: [foundry],
	transports: {
		// [mainnet.id]: http(),
		// [sepolia.id]: http(),
		[foundry.id]: http(),
	},
})
