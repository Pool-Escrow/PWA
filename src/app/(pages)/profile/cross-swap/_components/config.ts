// Configuration
export const CONFIG = {
    API: {
        BASE_URL: 'https://www.okx.com',
        RETRY: { MAX_ATTEMPTS: 3, DELAY: 1000 },
        HEADERS: {
            'Content-Type': 'application/json',
            'OK-ACCESS-KEY': process.env.NEXT_PUBLIC_OKX_API_KEY,
            'OK-ACCESS-PASSPHRASE': process.env.NEXT_PUBLIC_OKX_PASSPHRASE,
            'OK-ACCESS-PROJECT': process.env.NEXT_PUBLIC_OKX_PROJECT_ID,
            'OK-ACCESS-SECRET': process.env.NEXT_PUBLIC_OKX_SECRET_KEY,
        },
        ENDPOINTS: {
            'tokenPairs/crosschain': { path: '/api/v5/dex/cross-chain/supported/bridge-tokens-pairs', call: 'GET' },
            'tokens': { path: '/api/v5/dex/cross-chain/supported/tokens', call: 'GET' },
            'tokens/all': { path: '/api/v5/dex/aggregator/all-tokens', call: 'GET' },
            'chains': { path: '/api/v5/dex/cross-chain/supported/chain', call: 'GET' },
            'route': { path: '/api/v5/dex/cross-chain/quote', call: 'GET' },
            'route/detailed': { path: '/priapi/v1/dx/trade/bridge/v3/quote', call: 'GET' },
            'approve': { path: '/api/v5/dex/aggregator/approve-transaction', call: 'GET' },
            'swap': { path: '/api/v5/dex/cross-chain/build-tx', call: 'GET' },
            'status': { path: '/api/v5/dex/cross-chain/status', call: 'GET' },
            'approval/status': { path: '/priapi/v1/dx/trade/multi/batchGetTokenApproveInfo', call: 'GET' },
            'history': { path: '/api/v5/wallet/post-transaction/transactions-by-address', call: 'GET' },
            'orders': { path: '/priapi/v1/dx/trade/multi/orders', call: 'GET' },
            'account/OKX': { path: '/api/v5/wallet/account/create-wallet-account', call: 'POST' },
            'accounts': { path: '/api/v5/wallet/account/accounts', call: 'GET' },
        },
    },
    CHAIN: {
        BASE: {
            chainId: '8453',
            chainName: 'Base',
            platformId: 11,
            routerAddress: '0x57df6092665eb6058DE53939612413ff4B09114E' as const,
            nativeCurrency: {
                name: 'ETH',
                symbol: 'ETH',
                decimals: 18,
                address: '0x4200000000000000000000000000000000000006' as const,
            },
            tokens: {
                USDC: {
                    decimals: '6',
                    tokenContractAddress: '0x833589fcd6edb6e08f4c7c32d4f71b54bda02913',
                    dexTokenApproveAddress: '0x57df6092665eb6058DE53939612413ff4B09114E',
                    tokenName: 'usdc',
                    tokenSymbol: 'USDC',
                    tokenLogoUrl:
                        'https://static.okx.com/cdn/web3/currency/token/784-0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7::usdc::USDC-1.png/type=default_350_0?v=1735272019523',
                },
            },
            rpcUrls: ['https://mainnet.base.org'],
            blockExplorerUrls: ['https://basescan.org'],
        },
    },
} as const

// Tipos de ayuda para el CONFIG
export type ChainConfig = typeof CONFIG.CHAIN
export type APIConfig = typeof CONFIG.API
