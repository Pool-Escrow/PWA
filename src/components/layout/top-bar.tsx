'use client'

import { Button } from '@/components/ui/button'
import { Icons } from '@/components/ui/icons'
import UserMenu from '@/components/user/user-menu'
import usePrivacy from '@/hooks/use-privacy'
import BackButton from './back-button'

interface TopBarProps {
  backButton?: boolean | 'rounded'
  actionButton?: React.ReactNode
  title?: string
}

export default function TopBar({ backButton, actionButton, title }: TopBarProps) {
  const { hidden, toggle } = usePrivacy()

  return (
    <header className="z-30 bg-transparent text-white">
      <nav className="mx-auto h-[68px] max-w-screen-md px-safe-or-6">
        <div className="grid h-24 grid-cols-[1fr_auto_1fr] items-center">
          <div className="w-6">{backButton === 'rounded' ? <BackButton rounded /> : backButton && <BackButton />}</div>
          <div className="text-center text-[14px] font-medium text-black">{Boolean(title) && title}</div>
          <div className="flex items-center gap-4 justify-self-end">
            <Button
              size="icon"
              variant="ghost"
              className={`
                z-10 size-6 text-white
                hover:bg-transparent
              `}
              onClick={() => {
                void toggle()
              }}
            >
              {hidden
                ? <Icons.eyeOff className="size-6" />
                : <Icons.eye className="size-6" />}
            </Button>
            {actionButton ?? <UserMenu />}
          </div>
        </div>
      </nav>
    </header>
  )
}
