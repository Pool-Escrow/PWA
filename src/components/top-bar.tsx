'use client'

import { motion } from 'framer-motion'
import BackButton from './back-button'
import { usePathname } from 'next/navigation'
import { getPageTransition } from '@/lib/utils/animations'
import UserMenu from './user-menu'
import Image from 'next/image'
import type { StaticImport } from 'next/dist/shared/lib/get-img-props'
import eyeVisibleIcon from '@/public/app/icons/svg/eyevisible.svg'
import eyeInvisibleIcon from '@/public/app/icons/svg/eyeinvisible.svg'
import { Button } from '@/app/_components/ui/button'
import { useEncryptStore } from '@/app/_stores/encrypt'

export type TopBarProps = {
    backButton?: boolean
    actionButton?: React.ReactNode
    title?: string
}

function TopBarContent({ backButton, actionButton, title }: TopBarProps) {
    const pathname = usePathname()
    const pageTransition = getPageTransition(pathname)
    const { isEncoded, toggleEncryption } = useEncryptStore()

    return (
        <motion.div
            variants={pageTransition.variants}
            initial={false}
            animate='animate'
            exit='exit'
            key={pathname}
            transition={pageTransition.transition}
            className='grid h-24 grid-cols-[1fr_auto_1fr] items-center'>
            <div className='w-6'>{backButton && <BackButton />}</div>
            <div className='text-center'>{Boolean(title) && title}</div>
            <div className='flex items-center gap-4 justify-self-end'>
                <Button
                    size='icon'
                    variant='ghost'
                    className='z-10 size-6 text-white hover:bg-transparent'
                    onClick={toggleEncryption}>
                    <Image
                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
                        src={isEncoded ? eyeVisibleIcon : (eyeInvisibleIcon as StaticImport)}
                        alt={isEncoded ? 'Show balance' : 'Hide balance'}
                        className='size-6'
                    />
                </Button>
                {actionButton ?? <UserMenu />}
            </div>
        </motion.div>
    )
}

export default function TopBar(props: TopBarProps) {
    return (
        <header className='z-30 bg-transparent text-white'>
            <nav className='mx-auto h-[68px] max-w-screen-md px-safe-or-6'>
                <TopBarContent {...props} />
            </nav>
        </header>
    )
}
