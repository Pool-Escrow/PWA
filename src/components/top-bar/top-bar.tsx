'use client'

import { usePathname } from 'next/navigation'
import { getTopBarElements } from './top-bar.config'
import { useUserStore } from '@/stores/profile.store'
import { useEffect } from 'react'
import { getUserProfile } from '../profile/profile.action'

export default function TopBarLayout(): JSX.Element {
    const pathname = usePathname()
    const { left, center, right } = getTopBarElements(pathname)
    const { setProfile, setError } = useUserStore()

    useEffect(() => {
        async function loadUserProfile() {
            try {
                const profile = await getUserProfile()
                setProfile(profile)
            } catch (error) {
                setError(error as Error)
            }
        }

        loadUserProfile()
    }, [setProfile, setError])

    return (
        <header className='fixed left-0 top-0 z-30 w-dvw'>
            <nav className='flex-center mx-auto h-24 max-w-screen-md bg-white pt-5 px-safe-or-6 *:flex *:flex-1 *:items-center'>
                <div className='justify-start'>{left}</div>
                <div className='justify-center'>{center}</div>
                <div className='justify-end'>{right}</div>
            </nav>
        </header>
    )
}
