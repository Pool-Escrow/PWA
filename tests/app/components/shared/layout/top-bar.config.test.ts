import { getTopBarElements } from '@/components/top-bar/top-bar.config'

describe('getTopBarElements', () => {
    test('should return default elements for root path', () => {
        const elements = getTopBarElements('/')
        expect(elements).toMatchSnapshot()
    })

    test('should return elements for /my-pools', () => {
        const elements = getTopBarElements('/my-pools')
        expect(elements).toMatchSnapshot()
    })

    test('should return default elements for unknown paths', () => {
        const elements = getTopBarElements('/unknown/path')
        expect(elements).toMatchSnapshot()
    })

    test('should return elements for /profile/new', () => {
        const elements = getTopBarElements('/profile/new')
        expect(elements).toMatchSnapshot()
    })

    test('should return elements for /participant/[id]', () => {
        const elements = getTopBarElements('/participant/123')
        expect(elements).toMatchSnapshot()
    })

    test('should return elements for /pool/[poolId]', () => {
        const elements = getTopBarElements('/pool/456')
        expect(elements).toMatchSnapshot()
    })

    test('should return default elements for /participant/:id/some-other-path', () => {
        const elements = getTopBarElements('/participant/123/some-other-path')
        expect(elements).toMatchSnapshot()
    })

    test('static route should have priority over dynamic route', () => {
        const elementsForStaticRoute = getTopBarElements('/profile/new')
        const elementsForDynamicRoute = getTopBarElements('/participant/123')

        expect(elementsForStaticRoute).not.toEqual(elementsForDynamicRoute)
        expect(elementsForStaticRoute).toMatchSnapshot()
        expect(elementsForDynamicRoute).toMatchSnapshot()
    })

    // Add more tests as needed for additional routes
})
