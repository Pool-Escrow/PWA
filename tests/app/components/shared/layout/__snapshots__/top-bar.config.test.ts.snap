// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTopBarElements > should return default elements for /participant/:id/some-other-path 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": null,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > should return default elements for root path 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": null,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > should return default elements for unknown paths 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": null,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > should return elements for /my-pools 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > should return elements for /participant/[id] 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > should return elements for /participant/new 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LinkComponent)
    href="/"
  >
    Skip
  </ForwardRef(LinkComponent)>,
}
`;

exports[`getTopBarElements > should return elements for /pool/[poolId] 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LoadableComponent) />,
}
`;

exports[`getTopBarElements > static route should have priority over dynamic route 1`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LinkComponent)
    href="/"
  >
    Skip
  </ForwardRef(LinkComponent)>,
}
`;

exports[`getTopBarElements > static route should have priority over dynamic route 2`] = `
{
  "center": <ForwardRef(LoadableComponent) />,
  "left": <ForwardRef(LoadableComponent) />,
  "right": <ForwardRef(LoadableComponent) />,
}
`;
